### PR TITLE
SSI spousal deeming uses wrong FBR: couples lose /month when deeming applies

### LINKED ISSUE #6842: SSI spousal deeming uses wrong FBR: couples lose benefit when deeming applies
# Critical Bug: SSI Spousal Deeming Uses Individual FBR Instead of Couple FBR

## Executive Summary

There is a **critical bug** in the SSI spousal deeming implementation. When spousal deeming applies (ineligible spouse's gross income > $483/month), the code incorrectly uses the **individual FBR** ($967/month) instead of the **couple FBR** ($1,450/month) as required by 20 CFR §416.1163.

**This causes couples to receive $483/month ($5,796/year) LESS than they should when deeming applies.**

---

## How SSI Spousal Deeming Should Work

According to 20 CFR §416.1163 and SSA POMS SI 01320.400, there are three scenarios for married couples:

### Scenario A: Both Spouses Eligible
- **Condition:** Both are aged/blind/disabled
- **FBR Used:** Couple rate ($1,450/month)
- **Calculation:** Combine incomes, apply exclusions, subtract from couple FBR, divide by 2
- **Status:** ✅ Working correctly

### Scenario B: One Eligible, No Deeming
- **Condition:** One aged/blind/disabled, ineligible spouse's gross income ≤ $483/month
- **FBR Used:** Individual rate ($967/month)
- **Calculation:** Only eligible spouse's income counted
- **Status:** ✅ Working correctly

### Scenario C: One Eligible, Deeming Applies
- **Condition:** One aged/blind/disabled, ineligible spouse's gross income > $483/month
- **FBR Used:** **Couple rate ($1,450/month)**
- **Calculation:** Combine incomes, apply exclusions, subtract from couple FBR, cap at individual FBR
- **Status:** ❌ **BROKEN** - Uses individual rate instead of couple rate

---

### Current (Broken) Code in `ssi_amount_if_eligible.py`

```python
head_or_spouse_amount = where(
    person("ssi_claim_is_joint", period),  # Only TRUE when BOTH eligible
    p.couple / 2,
    p.individual,  # ❌ WRONG for deeming scenarios
)
```

### What's Wrong

The variable `ssi_claim_is_joint` only returns `True` when **both spouses are aged/blind/disabled**.

When only one spouse is eligible:
- `ssi_claim_is_joint` = `False`
- Code uses `p.individual` ($967)
- This is correct for Scenario B (no deeming)
- **But wrong for Scenario C (deeming applies)** - should use `p.couple` ($1,450)

The code fails to distinguish between Scenario B and Scenario C, always using the individual rate for both.

---

## Understanding the Deeming Threshold

**Key regulatory point:** The $483 threshold compares the spouse's **GROSS income** (after child allocations only), **NOT** the countable income after $20/$65 exclusions.

### Deeming Decision Process:
1. Get ineligible spouse's gross income
2. Subtract child allocations (if any)
3. **Compare to $483 threshold**
   - If ≤ $483: No deeming (use individual FBR)
   - If > $483: Deeming applies (use couple FBR)
4. If deeming applies, **then** apply $20/$65 exclusions to combined income
5. Calculate benefit and cap at individual FBR

---

### Example 1: Spouse Earns $1,100/month

**Family:**
- Woman: Healthy, earns $1,100/month gross
- Man: Disabled, $0 income
- No children

**Does deeming apply?**
- Woman's gross: $1,100 > $483 → **YES**

**Combined countable:**
- Combined: $1,100/month
- Exclusions: $1,100 - $20 - $65 = $1,015
- Half: $1,015 / 2 = $507.50/month

**Before PR (WRONG):**
- Uses individual FBR: $967/month
- SSI = $967 - $507.50 = **$459.50/month**

**After PR (CORRECT):**
- Uses couple FBR: $1,450/month
- SSI = $1,450 - $507.50 = **$942.50/month**

**Impact: +$483/month (+$5,796/year)**

---

### Example 2: Spouse Earns $967/month (Cap Example)

**Family:**
- Woman: Healthy, earns $967/month gross
- Man: Disabled, $0 income

**Does deeming apply?**
- Woman's gross: $967 > $483 → **YES**

**Combined countable:**
- $967 - $20 - $65 = $882
- Half: $882 / 2 = $441/month

**Before PR (WRONG):**
- Individual FBR: $967
- SSI = $967 - $441 = **$526/month**

**After PR (CORRECT):**
- Couple FBR: $1,450
- Benefit before cap: $1,450 - $441 = $1,009
- **Cap at individual FBR:** min($1,009, $967) = **$967/month**

**Impact: +$441/month (+$5,292/year)**

**Note:** This demonstrates why the individual FBR cap is necessary - when gross income is just above $483, exclusions reduce countable income enough that the uncapped benefit would exceed the individual rate.

---

### Example 3: With Child Allocation

**Family:**
- Woman: Healthy, earns $3,000/month
- Man: Disabled, $0 income
- Child: Healthy, $0 income

**Child allocation:**
- $1,450 - $967 = $483/month
- Woman's gross after: $3,000 - $483 = $2,517/month

**Does deeming apply?**
- $2,517 > $483 → **YES**

**Combined countable:**
- $2,517 - $20 - $65 = $2,432
- Half: $2,432 / 2 = $1,216/month

**Before PR (WRONG):**
- Individual FBR: $967
- Benefit: $967 - $1,216 = -$849 → **$0**

**After PR (CORRECT):**
- Couple FBR: $1,450
- Benefit: $1,450 - $1,216 = **$234/month**

**Impact: +$234/month (+$2,808/year)**

---

### Example 4: No Deeming (Low Income)

**Family:**
- Woman: Healthy, earns $400/month
- Man: Disabled, $0 income

**Does deeming apply?**
- Woman's gross: $400 ≤ $483 → **NO**

**Before and After PR: SAME**
- Uses individual FBR: $967
- Man's countable: $0
- SSI = $967 - $0 = **$967/month**

**Impact: $0** (no change - this scenario was already correct)

---

## Comparison Table: Before vs. After PR

| Woman's Earned | Deeming? | Before PR (Wrong) | After PR (Correct) | Benefit Gained |
|----------------|----------|-------------------|-------------------|----------------|
| $400 | No | $967 | $967 | $0 |
| $500 | Yes | $742 | $1,225 | **+$483** |
| $967 | Yes | $526 | $967 (capped) | **+$441** |
| $1,100 | Yes | $459.50 | $942.50 | **+$483** |
| $1,500 | Yes | $242.50 | $725.50 | **+$483** |
| $2,000 | Yes | $9.50 | $492.50 | **+$483** |
| $3,000 (no child) | Yes | $0 | $0 | $0 |
| $3,000 (with child) | Yes | $0 | $234 | **+$234** |

**Key pattern:** When deeming applies, couples gain $483/month except when:
- Income is just above threshold (gain varies due to cap)
- Income is so high that even couple FBR results in $0

---

### 1. New Helper Variable: `is_ssi_spousal_deeming.py`
Determines when spousal deeming applies:
```python
# Check if spouse's gross income (after child allocations) > $483
spouse_deemed_income = add(person, period, [
    "ssi_earned_income_deemed_from_ineligible_spouse",
    "ssi_unearned_income_deemed_from_ineligible_spouse",
])
diff = (p.couple - p.individual) * MONTHS_IN_YEAR  # $5,796/year
return is_eligible_individual & (spouse_deemed_income > diff)
```

### 2. Fixed `ssi_amount_if_eligible.py`
```python
is_joint_claim = person("ssi_claim_is_joint", period)
deeming_applies = person("is_ssi_spousal_deeming", period)

individual_or_deeming_amount = where(
    deeming_applies,
    p.couple,  # ✅ Use couple rate when deeming!
    p.individual,
)

head_or_spouse_amount = where(
    is_joint_claim,
    p.couple / 2,
    individual_or_deeming_amount,
)
```

### 3. Added Cap in `ssi.py`
Moved cap logic from `uncapped_ssi` to `ssi` for better separation of concerns:
```python
uncapped = person("uncapped_ssi", period)

# Apply floor: cannot be negative
benefit = max_(0, uncapped)

# Apply cap when deeming: cannot exceed individual FBR
deeming_applies = person("is_ssi_spousal_deeming", period)
individual_max = p.individual * MONTHS_IN_YEAR
capped_benefit = min_(benefit, individual_max)

return where(
    deeming_applies,
    capped_benefit,
    benefit,
)
```

### 4. Refactoring Improvements

- Eliminated duplicate threshold checking by reusing `is_ssi_spousal_deeming` in `ssi_income_deemed_from_ineligible_spouse.py`
- Removed nested `where()` statements for better readability
- Better variable naming for clarity

---

## Files Modified

1. **New:** `policyengine_us/variables/gov/ssa/ssi/eligibility/income/deemed/from_ineligible_spouse/is_ssi_spousal_deeming.py`
2. **Modified:** `policyengine_us/variables/gov/ssa/ssi/ssi_amount_if_eligible.py`
3. **Modified:** `policyengine_us/variables/gov/ssa/ssi/ssi.py`
4. **Modified:** `policyengine_us/variables/gov/ssa/ssi/eligibility/income/deemed/from_ineligible_spouse/ssi_income_deemed_from_ineligible_spouse.py`
5. **Modified:** `policyengine_us/tests/policy/baseline/gov/ssa/ssi/integration_tests.yaml`

---

## Testing

All 105+ SSI tests pass with the fix, including:
- Historical CFR regulation examples (1986)
- Modern 2025 test scenarios
- Spousal deeming edge cases
- Parent-to-child deeming scenarios
- Integration tests with various family structures
- New test cases demonstrating cap behavior and child allocation impact

---

## Regulatory Compliance

This fix brings PolicyEngine into compliance with:
- **20 CFR §416.1163** - Deeming of income from an ineligible spouse
- **SSA POMS SI 01320.400** - Operational guidance for spousal deeming

The implementation now correctly follows the SSA's step-by-step deeming calculation process.

### PR DESCRIPTION
**Pull request by hua7450**
Fixes #6842

### CODE REVIEW

File: policyengine_us/variables/gov/ssa/ssi/eligibility/income/deemed/from_ineligible_spouse/is_ssi_spousal_deeming.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    When deeming applies, the benefit calculation uses the couple FBR instead
+    of the individual FBR, recognizing that two people have higher expenses.
+    """
+
+    def formula(person, period, parameters):
+        # Only applies to eligible individuals (not eligible spouses in a couple)
+        is_eligible_individual = person("is_ssi_eligible_individual", period)
+
+        # Get spouse's leftover income (after child allocations)
+        spouse_earned = person(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/ssa/ssi/eligibility/income/deemed/from_ineligible_spouse/is_ssi_spousal_deeming.py]:
```suggestion
        spouse_deemed_earned_income = person(
```

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/ssa/ssi/eligibility/income/deemed/from_ineligible_spouse/is_ssi_spousal_deeming.py]: Can we be a bit descriptive in the variable names - just to make sure we accurately represent everything?

File: policyengine_us/variables/gov/ssa/ssi/ssi_amount_if_eligible.py [hunk 2]
```diff
@@ -12,12 +12,33 @@ class ssi_amount_if_eligible(Variable):
...
-            person("ssi_claim_is_joint", period),
-            p.couple / 2,
-            p.individual,
+            is_joint_claim,
+            p.couple / 2,  # Scenario 1: Both eligible
+            where(
+                deeming_applies,
+                p.couple,  # Scenario 3: Deeming applies - use couple rate!
+                p.individual,  # Scenario 2: No deeming
+            ),
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/ssa/ssi/ssi_amount_if_eligible.py]: Lets break out

File: policyengine_us/tests/policy/baseline/gov/ssa/ssi/integration_tests.yaml [hunk 3]
```diff
@@ -68,10 +68,12 @@
...
     is_ssi_eligible: [true, false, true]
     # Spouse's countable: $2,000 - $20 - $65 - $957.50 = $957.50/month = $11,490/year
-    # PolicyEngine deems all of spouse's countable income (doesn't subtract spouse allowance)
+    # Since $11,490 > $5,796 (couple - individual FBR diff), deeming applies
+    # With deeming, use couple FBR: $1,450/month = $17,400/year
     ssi_income_deemed_from_ineligible_spouse: [11_490, 0, 0]
     ssi_countable_income: [11_490, 0, 0]
-    ssi: [114, 0, 11_604]  # Adult1: $11,604 - $11,490 = $114; Child gets full FBR
+    # Adult1 SSI: $17,400 (couple FBR) - $11,490 (countable) = $5,910
+    ssi: [5_910, 0, 11_604]  # Adult1: $5,910 with couple rate; Child gets full FBR
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/ssa/ssi/integration_tests.yaml:76]: Big diff to an existing unit test. Why?

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/tests/policy/baseline/gov/ssa/ssi/integration_tests.yaml (hunk 3 above)
Comment (hua7450) [reply policyengine_us/tests/policy/baseline/gov/ssa/ssi/integration_tests.yaml:76]: Because the old code is wrong, check issue page of this PR. 
Screenshot 2025-11-19 at 3 33 35 PM

File: policyengine_us/tests/policy/baseline/gov/ssa/ssi/integration_tests.yaml [hunk 4]
```diff
@@ -217,12 +219,14 @@
...
-    # PolicyEngine deems all countable income without subtracting spouse allowance
+    # Since $29,490 > $5,796 (couple - individual FBR diff), deeming applies
+    # With deeming, use couple FBR: $1,450/month = $17,400/year
+    # Adult1 SSI: $17,400 (couple FBR) - $29,490 (countable) = -$12,090, capped at $0
     ssi_income_deemed_from_ineligible_spouse: [29_490, 0, 0]
     # Parent income deeming to child also gets high amount
     ssi_unearned_income_deemed_from_ineligible_parent: [0, 0, 17_886]
     ssi_countable_income: [29_490, 0, 17_646]
-    ssi: [0, 0, 0]  # Both disabled members have too much deemed income
+    ssi: [0, 0, 0]  # Both disabled members have countable income exceeding FBR
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/ssa/ssi/integration_tests.yaml:229]: could we add one more test that was breaking before and works now? With an explnantion why it was breaking

Comment (hua7450) [reply policyengine_us/tests/policy/baseline/gov/ssa/ssi/integration_tests.yaml:229]: Case 8, before this PR, it would return 526. It use individual rate - countable income. 
In this PR, case 8 return 967, it use couple rate - countable income, then cap at individual rate.

File: policyengine_us/variables/gov/ssa/ssi/eligibility/income/deemed/from_ineligible_spouse/is_ssi_spousal_deeming.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+from policyengine_us.model_api import *
+
+
+class is_ssi_spousal_deeming(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/ssa/ssi/eligibility/income/deemed/from_ineligible_spouse/is_ssi_spousal_deeming.py]:
```suggestion
class is_ssi_spousal_deeming_applies(Variable):
```

**Review by MaxGhenis — APPROVED** [review]
This PR correctly implements the SSI spousal deeming rules per 20 CFR §416.1163 and SSA POMS SI 01320.400.

Key points:
- Correctly uses the couple FBR when deeming applies.
- Accurately implements the deeming threshold (spouse's income > difference between couple and individual FBR).
- Properly caps the benefit at the individual FBR when deeming applies.

Comprehensive tests cover various scenarios, including the cap and child allocations. The code is well-refactored and addresses the bug effectively.
